src/rez/package_cache.py

```diff
@@ -130,6 +132,65 @@ def get_cached_root(self, variant):
 
         return rootpath
 
+    def get_variant_size(self, variant_root):
+        """Get the size of the variant root.
+
+        Args:
+            variant_root: The rez resolved variant root.
+
+        Returns:
+            int: The size in bytes of the variant root (may exceed buffer slightly).
+        """
+        variant_size = 0
+
+        with os.scandir(variant_root) as variant_scan:
+            for variant_dir in variant_scan:
+                try:
+                    if variant_dir.is_file(follow_symlinks=False):
+                        variant_size += variant_dir.stat(follow_symlinks=False).st_size
+                    elif variant_dir.is_dir(follow_symlinks=False):
+                        variant_size += self.get_variant_size(variant_dir.path)
+                except OSError:
+                    pass
+                except FileNotFoundError:
+                    pass
+
+        return variant_size
+
+    def cache_near_full(self):
+        """ Get the cache available space
+
+        Returns:
+            bool: True if available cache space is below buffer, otherwise False.
+        """
+        _, _, free = shutil.disk_usage(self.path)
+        return (free < config.package_cache_space_buffer)
+
+    def variant_not_cachable(self, rez_variant_root):
```

Comment:
I think I would call this function `variant_meets_space_requirements` or something similar, since there are potentially other reasons that a variant might not be cachable (such as being configured not to be cached)